### PR TITLE
[MNT] Complete test for sparse dataset

### LINKED ISSUE #1644: [MNT] Complete Test for Sparse Dataset Row ID and Ignore Attributes Inclusion
The test `test_get_sparse_dataset_rowid_and_ignore_and_target` in [test_dataset.py#L382](https://github.com/openml/openml-python/blob/main/tests/test_datasets/test_dataset.py#L382) currently sets `ignore_attribute` and `row_id_attribute` on a sparse dataset but calls `get_data()` with `include_row_id=False` and `include_ignore_attribute=False`. This means the test doesn't verify behavior when these attributes are actually included in the output.

**Steps:**
1. Add a new test case or extend the existing one to call `get_data(target="class", include_row_id=True, include_ignore_attribute=True)`.
2. Assert the correct output shape (should exclude the ignored columns and include row IDs if applicable).
3. Verify that categorical indicators are updated accordingly.
4. Run the test to ensure it passes with the sparse dataset.

The test should validate that sparse datasets handle row ID and ignore attributes correctly when included, improving test coverage for edge cases.

Source: Mentioned TODO in source code (same line)

### PR DESCRIPTION
**Pull request by omkar-334**
Fixes #1644
 
1. Added coverage for sparse dataset get_data when row_id and ignore attributes are included.   
2. Split behavior into two tests: `test_get_sparse_dataset_excludes_rowid` (checks 19998 columns) and `test_get_sparse_dataset_includes_rowid` (checks 20000 columns). 
3. Updated test names to make include vs exclude behavior clear.

cc @geetu040 @PGijsbers

### CODE REVIEW

**Comment by omkar-334** [comment]
Screenshot 2026-02-19 at 12 27 14 AM

File: tests/test_datasets/test_dataset.py [hunk 1]
```diff
@@ -397,6 +396,21 @@ def test_get_sparse_dataset_rowid_and_ignore_and_target(self):
...
+            include_row_id=True,
+            include_ignore_attribute=True,
+        )
+        assert all(dtype == pd.SparseDtype(np.float32, fill_value=0.0) for dtype in X.dtypes)
+        assert isinstance(y.dtypes, pd.SparseDtype)
+        assert X.shape == (600, 20000)
+        assert len(categorical) == 20000
+        self.assertListEqual(categorical, [False] * 20000)
+        assert y.shape == (600,)
+
```
[the remark this answers is not in the document]
Comment (omkar-334) [reply tests/test_datasets/test_dataset.py:413]: @geetu040 do you think we need the other 2 scenarios as well?
